### PR TITLE
Lra 250 mi classrooms capacity range add validation

### PR DESCRIPTION
**Pull request by britaumich**
Added a stimulus controller function to validate capacity (min <= max).
If not valid, a message "Min should be smaller than Max" is displayed under form fields for min and max capacity, and the form is not submitted.

### CODE REVIEW

File: app/packs/controllers/autosubmit_controller.js [hunk 1]
```diff
@@ -22,6 +22,31 @@ export default class extends Controller {
     }, 0)
   }
 
+  capacitySubmit(event) {
+    clearTimeout(this.timeout)
+
+    this.timeout = setTimeout(() => {
```
Comment (anantaUM) [inline app/packs/controllers/autosubmit_controller.js:28]: Why do we have a timeout of 0.6 seconds ?

File: app/packs/controllers/autosubmit_controller.js [hunk 2]
```diff
@@ -22,6 +22,31 @@ export default class extends Controller {
...
+      var min_capacity = parseInt(this.min_capacityTarget.value)
+      var max_capacity = parseInt(this.max_capacityTarget.value)
+
+      if (min_capacity > max_capacity) {
+        this.capacity_errorTarget.classList.add("capacity-error--display")
+        this.capacity_errorTarget.classList.remove("capacity-error--hide")
+        this.capacity_errorTarget.innerText = "Min should be smaller than Max"
+        event.preventDefault()
```
Comment (anantaUM) [inline app/packs/controllers/autosubmit_controller.js:36]: Which default behavior are we trying to stop here ?

File: app/packs/controllers/autosubmit_controller.js [hunk 3]
```diff
@@ -22,6 +22,31 @@ export default class extends Controller {
...
 
+  capacitySubmit(event) {
+    clearTimeout(this.timeout)
+
+    this.timeout = setTimeout(() => {
+      var min_capacity = parseInt(this.min_capacityTarget.value)
+      var max_capacity = parseInt(this.max_capacityTarget.value)
+
+      if (min_capacity > max_capacity) {
+        this.capacity_errorTarget.classList.add("capacity-error--display")
```
Comment (anantaUM) [inline app/packs/controllers/autosubmit_controller.js:33]: It is possible that this class is already present in the ClassList. Do we need some conditional check or ruby takes care of it ?

File: app/packs/controllers/autosubmit_controller.js (hunk 2 above)
Comment (britaumich) [reply app/packs/controllers/autosubmit_controller.js:36]: By default, the form will be submitted. I think we shouldn't submit a form if min capacity > max capacity. If submitted the search will return an empty result.

File: app/packs/controllers/autosubmit_controller.js (hunk 3 above)
Comment (britaumich) [reply app/packs/controllers/autosubmit_controller.js:33]: The span for the error is on the page, and it's hidden by default:
`<span class="capacity-error--hide text-red-700 text-sm mt-2 -mb-2" data-autosubmit-target="capacity_error">`

capacity-error--hide class in application.sass:
`.capacity-error--hide
  @apply hidden`

The autosubmit_controller.js is changing this class to [capacity-error--display] to display the error if there is an error, or to [capacity-error--hide] if there is no error. 

Please let me know if my explanations are not clear and if you have more questions.

Comment (anantaUM) [reply app/packs/controllers/autosubmit_controller.js:32]: As per follow up discussion with Rita, the class gets added only once even though it is already present inn the list.

**Review by anantaUM — APPROVED** [review]

File: app/packs/controllers/autosubmit_controller.js (hunk 2 above)
Comment (anantaUM) [reply app/packs/controllers/autosubmit_controller.js]: Based on follow up discussion with Rita, this line of code is not required and was removed.

**Review by anantaUM — APPROVED** [review]

**Pull request merged**